src/config.c

```diff
@@ -2970,6 +2970,26 @@ void rewriteConfigLatencyTrackingInfoPercentilesOutputOption(standardConfig *con
     rewriteConfigRewriteLine(state,name,line,1);
 }
 
+static int applyClientMaxMemoryUsage(const char **err) {
+    UNUSED(err);
+    listIter li;
+    listNode *ln;
+    /* When client eviction is enabled update memory buckets for all clients.
+     * When disabled, clear that data structure. */
+    listRewind(server.clients, &li);
+    while ((ln = listNext(&li)) != NULL) {
```

Comment:
i just realized that this scan is running on every config change, even if changing it from 30% to 29%.
i guess we have to somehow cache the previous value of that config and skip it in such cases.

Reply:
instead of adding another server variable, i decided to try to use server.client_mem_usage_buckets as an indication of the previous config.
let me know what you think, shall we revert that and just add another var to hold the active config value?

Reply:
The suggested solution sounds good to me.

Reply:
thanks, did you run another set of eyes on the changes i committed?
please also ack that my edits on the top comment are ok (or enhance / fix them).
then i guess we'll merge, and also backport to 7.0.6 which i wanna release soon.

Reply:
I was not able to approve this change earlier because I started the pull request.

Both, the code changes and the PR description looks good to me.